src/gtMerger/matchAndMerge.py

```diff
@@ -0,0 +1,159 @@
+
+    # Create path OS independent for csv file
+    csv_path = os.path.join(args.dataDir, "travel-behavior.csv")
+    # Load OBA data
+    oba_data = pd.read_csv(csv_path)
+    # Preprocess OBA data
+    oba_data = preprocessObaData(oba_data, args)
+    print("OBA data info()\n")
+    print(oba_data.info())
+
+    # Data preprocessing is over
+    merged_data_frame = pd.merge_asof(gt_data, oba_data, on="ClosestTime", direction="nearest", tolerance=pd.Timedelta(str(args.tolerance)+"ms"))
+    print(merged_data_frame.head())
+
+    # Save to csv
+    merged_file_path = os.path.join(args.dataDir, "output", "mergedData.csv")
+    merged_data_frame.to_csv(path_or_buf =merged_file_path, index=False)
+    
+
+def preprocessObaData(data_csv, args):
+    """ Preprocess the csv data file from oba-firebase-export as follows:
+        - Change activity start date datatype from str to datetime
+        - Drop observations whose activity start date are NaN after data type conversion
+        - Drop observations that does not mathc the time span and distance requirements from args
+          minActivitySpan and minTripLength
+        - Add column required to be used as key while merging with ground truth data
+    
+    Args:
+        data_csv: A data frame loaded from a csv file genrated by oba-firebase-export
+        args: List of arguments from command prompt
+
+    Returns:
+        Preprocessed dataframe
+    """
+    # Change Activity Start Date and Time* (UTC) to datetime
+    data_csv['Activity Start Date and Time* (UTC)']= pd.to_datetime(data_csv['Activity Start Date and Time* (UTC)'], errors='coerce', utc=True)
+    
+    # Drop NaN rows for relevant columns
+    clean_data_csv = data_csv.dropna(subset= ['Activity Start Date and Time* (UTC)', 'Origin location Date and Time (*best) (UTC)', 
+    'Duration* (minutes)', 'Origin-Destination Bird-Eye Distance* (meters)'])
+
+    # Remove trips with Duration less than args.minActivitySpan mins and distance les than args.minTripLength
+    clean_data_csv = clean_data_csv[(data_csv['Duration* (minutes)']>=args.minActivitySpan) & (data_csv['Origin-Destination Bird-Eye Distance* (meters)']>=args.minTripLength)]
+
+    # Add the data to be dropped to a data frame
+    data_csv_dropped = pd.merge(data_csv, clean_data_csv, how='outer', indicator=True).query("_merge != 'both'").drop('_merge', axis=1).reset_index(drop=True)
+
+    # Save data to be dropped to a csv file
+    dropped_file_path = os.path.join(args.dataDir, "logs", "droppedObaData.csv")
+    data_csv_dropped.to_csv(path_or_buf = dropped_file_path, index=False)
+    
+    # Add column to be used in merge_asoft
+    clean_data_csv['ClosestTime'] = clean_data_csv['Activity Start Date and Time* (UTC)']
+    
+    return clean_data_csv
+
+def preprocessGtData(gt_data):
+    """ Preprocess the ground Truth xlsx data file as follows:
+        - Remove unnmed coulmns if exist
+        - Replace ? by 0 to correct possible not rounded seconds
```

Comment:
We shouldn't need to handle `?` anymore, right, now that we have a separate column indicating if the data is rounded or not? If the data shouldn't include the `?`, I'd remove the processing of it.

Reply:
I will push this change in the first version since I am still using it to perform some local test.